build.ps1

```diff
@@ -1,81 +1,101 @@
-
-  if(($LASTEXITCODE -ne 0) -Or ((Test-Path Env:\APPVEYOR) -eq $true))
-  {
-    Write-Host "Dotnet CLI not found - downloading latest version"
-
-    # Prepare the dotnet CLI folder
-    $env:DOTNET_INSTALL_DIR="$(Convert-Path "$PSScriptRoot")\.dotnet\win7-x64"
-    if (!(Test-Path $env:DOTNET_INSTALL_DIR))
-    {
-      mkdir $env:DOTNET_INSTALL_DIR | Out-Null
-    }
-
-    # Download the dotnet CLI install script
-    if (!(Test-Path .\dotnet\install.ps1))
-    {
-      Invoke-WebRequest "https://raw.githubusercontent.com/dotnet/cli/rel/1.0.0/scripts/obtain/dotnet-install.ps1" -OutFile ".\.dotnet\dotnet-install.ps1"
-    }
-
-    # Run the dotnet CLI install
-    & .\.dotnet\dotnet-install.ps1
-
-    # Add the dotnet folder path to the process. This gets skipped
-    # by Install-DotNetCli if it's already installed.
-    Remove-PathVariable $env:DOTNET_INSTALL_DIR
-    $env:PATH = "$env:DOTNET_INSTALL_DIR;$env:PATH"
-
-  }
-}
-
-function Remove-PathVariable
-{
-  [cmdletbinding()]
-  param([string] $VariableToRemove)
-  $path = [Environment]::GetEnvironmentVariable("PATH", "User")
-  $newItems = $path.Split(';') | Where-Object { $_.ToString() -inotlike $VariableToRemove }
-  [Environment]::SetEnvironmentVariable("PATH", [System.String]::Join(';', $newItems), "User")
-  $path = [Environment]::GetEnvironmentVariable("PATH", "Process")
-  $newItems = $path.Split(';') | Where-Object { $_.ToString() -inotlike $VariableToRemove }
-  [Environment]::SetEnvironmentVariable("PATH", [System.String]::Join(';', $newItems), "Process")
-}
-
-function Restore-Packages
-{
-    param([string] $DirectoryName)
-    & dotnet restore -v Warning ("""" + $DirectoryName + """")
-}
-
-function Test-Projects
-{
-    & dotnet test -c Release;
-    if($LASTEXITCODE -ne 0)
-    {
-      exit 3
-    }
-}
-
-########################
-# THE BUILD!
-########################
-
-Push-Location $PSScriptRoot
-
-# Install Dotnet CLI
-Install-Dotnet
-
-# Package restore
-Write-Host "Running package restore"
-Get-ChildItem -Path . -Filter *.xproj -Recurse | ForEach-Object { Restore-Packages $_.DirectoryName }
-
-# Tests
-Write-Host "Running tests"
-Get-ChildItem -Path .\test -Filter *.xproj -Exclude Nancy.ViewEngines.Razor.Tests.Models.xproj -Recurse | ForEach-Object {
-    Push-Location $_.DirectoryName
-    Test-Projects
-    Pop-Location
-}
-
-Pop-Location
+[CmdletBinding()]
+Param(
+    [string]$Target = "Default",
+    [ValidateSet("Quiet", "Minimal", "Normal", "Verbose", "Diagnostic")]
+    [string]$Verbosity = "Verbose",
+    [switch]$WhatIf,
+    [string]$NuGetSource = $null,
+    [string]$NuGetApiKey = $null,
+    [string]$Version = $null,
+    [switch]$SkipClean,
+    [switch]$SkipTests
+)
+
+######################################################################################################
+
+Function Install-Dotnet()
+{
+  $GlobalDotNetFound = &where.exe dotnet 2>&1 | Out-Null
+  $LocalDotNetFound = Test-Path (Join-Path $PSScriptRoot ".dotnet")
```

Comment:
Should make sure `$PSScriptRoot` has been set properly to work with older versions of PowerShell.